src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DocumentationDescriptor.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+internal abstract partial class DocumentationDescriptor : IEquatable<DocumentationDescriptor>
```

Comment:
It feels like the main thing blocking this from being a record is the nested array, right? Would it be worth introducing some struct that wraps array and making that sequence-equatable, and then this and the derived types could be record types?

Reply:
I'm not sure it's worth it for equality on two properties. Out of curiosity, do records cache hash code values? I'm guessing no?

Reply:
> Out of curiosity, do records cache hash code values

That's correct, they do not. Is that a significant source of speedup for this PR?

Reply:
It's not something I'd give up in order to make this a record. Eventually, serialization will use this as a key in a map. So, caching the hash code _will_ affect performance though the amount is likely to be small. However, sInce there are only two properties, I'm not seeing a lot of value in contorting this into a record. Since I'd have to provide a struct-based array wrapped, making it a record would use more code than leaving it as a normal class.